### PR TITLE
EN-2348: add guards when schema generator fails

### PR DESCRIPTION
**Pull request by JonathanLoscalzo**
## What changed?
- add tests for schema generations
- add try-except to skip errors during generation
- log those errors
- remove aws_config from AWSAccount
- change typing of organizations at AWSAccount

## Rationale
We need testing to check the schema generator for IAMbic's docs. In some cases, some models have issues to be generated because of circular dependencies or forward_refs not loaded (pydantic).

## How was it tested?
If it was manually verified, list the instructions for your reviewers to follow.
- [x] Unit Tests
- [ ] Functional Tests
- [x] Manually Verified

### CODE REVIEW

**Comment by JonathanLoscalzo** [comment]
@castrapel, test for AWSConfig is not working, so is just the only model that doesn't work. 
Let me know if you want to comment it out or what action we should take.

File: iambic/plugins/v0_1_0/aws/models.py [hunk 1]
```diff
@@ -489,7 +486,7 @@ async def set_account_organization_details(
         when the account is the organization account
         """
         self.organization = organization
-        self.aws_config = config
+        # self.aws_config = config
```
Comment (castrapel) [inline iambic/plugins/v0_1_0/aws/models.py:489]: Comment can be removed?

File: iambic/plugins/v0_1_0/aws/models.py [hunk 2]
```diff
@@ -568,7 +565,8 @@ def all_identifiers(self) -> set[str]:
     @property
     def organization_account(self) -> bool:
         """if current account is an organization account"""
-        return bool(self.organization and self.aws_config)
+        # return bool(self.organization and self.aws_config)
```
Comment (castrapel) [inline iambic/plugins/v0_1_0/aws/models.py:568]: Comment can be removed?

**Review by castrapel — APPROVED** [review]
LGTM

**Pull request merged**